### PR TITLE
Add RTP packet Queue

### PR DESCRIPTION
**Pull request by moninom1**
*Issue #, if available:*

*Description of changes:*
Add RTP packet Queue which will store all the outgoing RTP packets.
Add unit test cases to validate functionality of RTP packet Queue.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by aggarg — APPROVED** [review]

File: source/rtp_pkt_queue.c [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+/*----------------------------------------------------------------------------*/
+
+/**
+ * @brief Read and remove the oldest RTP packet info from the queue.
+ */
+RtpPacketQueueResult_t RtpPacketQueue_Dequeue( RtpPacketQueue_t * pQueue,
+                                               RtpPacketInfo_t * pRtpPacketInfo )
+{
+    RtpPacketQueueResult_t result = RTP_PACKET_QUEUE_RESULT_OK;
```
Comment (ActoryOu) [inline source/rtp_pkt_queue.c]: Is it possible to merge common part in `RtpPacketQueue_Dequeue()` and `RtpPacketQueue_Peek()` as they're pretty similar?

File: test/unit-test/rtp_packet_queue/rtp_packet_queue_utest.c [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    result = RtpPacketQueue_ForceEnqueue( &( rtpPacketQueue ),
+                                          &( rtpPacketInfo ),
+                                          &( deletedRtpPacketInfo ) );
+
+    TEST_ASSERT_EQUAL( RTP_PACKET_QUEUE_RESULT_OK, result );
+
+    result = RtpPacketQueue_Dequeue( &( rtpPacketQueue ),
+                                     &( rtpPacketInfo ) );
+
+    TEST_ASSERT_EQUAL( 10, rtpPacketInfo.seqNum );
```
Comment (ActoryOu) [inline test/unit-test/rtp_packet_queue/rtp_packet_queue_utest.c:201]:
```suggestion
    TEST_ASSERT_EQUAL( RTP_PACKET_QUEUE_RESULT_OK, result );
    TEST_ASSERT_EQUAL( 10, rtpPacketInfo.seqNum );
```
I suggest to check the result here.

File: test/unit-test/rtp_packet_queue/rtp_packet_queue_utest.c [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    result = RtpPacketQueue_ForceEnqueue( &( rtpPacketQueue ),
+                                          &( rtpPacketInfo ),
+                                          &( deletedRtpPacketInfo ) );
+
+    TEST_ASSERT_EQUAL( RTP_PACKET_QUEUE_RESULT_OK, result );
+
+    result = RtpPacketQueue_Peek( &( rtpPacketQueue ),
+                                  &( rtpPacketInfo ) );
+
+    TEST_ASSERT_EQUAL( 20, rtpPacketInfo.seqNum );
```
Comment (ActoryOu) [inline test/unit-test/rtp_packet_queue/rtp_packet_queue_utest.c:268]:
```suggestion
    TEST_ASSERT_EQUAL( RTP_PACKET_QUEUE_RESULT_OK, result );
    TEST_ASSERT_EQUAL( 20, rtpPacketInfo.seqNum );
```

File: source/rtp_pkt_queue.c (hunk 1 above)
Comment (moninom1) [reply source/rtp_pkt_queue.c]: RtpPacketQueue_Dequeue is removing the element from the queue , where as RtpPacketQueue_Peek is just returning the element. As the functionalities are different hence we are keeping both. 
But at later point of time while implementing application side code if it is not required we can remove it then.

Comment (ActoryOu) [reply source/rtp_pkt_queue.c]: Hi @moninom1,
I didn't mean to merge two API into one. What I mean is to merge the "common part" inside. What do you think?
For example:
```
RtpPacketQueue_Peek()
{
    /* Peek the element. */
}

RtpPacketQueue_Dequeue()
{
    RtpPacketQueue_Peek();

    /* removing that element from the queue. */
}
```

Comment (hassanctech) [reply source/rtp_pkt_queue.c]: Yes I agree with @ActoryOu .  I see also right now there is not consistent behavior in these two APIs if the packet info struct is NULL.  IN peek we will return bad param but in dequeue we will still update the readIndex and packetCount on the queue even when the packet info is NULL.  Do we intend for this divergence?

**Review by shubnil — APPROVED** [review]

File: source/rtp_pkt_queue.c (hunk 1 above)
Comment (moninom1) [reply source/rtp_pkt_queue.c]: Sure, updated.

**Review by ActoryOu — APPROVED** [review]

**Review by hassanctech — APPROVED** [review]
lgtm!

**Pull request merged**